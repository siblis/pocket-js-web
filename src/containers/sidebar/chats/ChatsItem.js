import React from 'react';
import { NavLink } from 'react-router-dom';

import styles from './ChatsItem.module.css';

const ChatsItem = props => (
    <div id="ChatsItem" className={styles.Block} onClick={() => props.handleClicks(props.id)}>
        <NavLink to={`/chats/${props.id}`}>
            <div className={styles.Item}>
                <div className={styles.Img}>
                    <img src="http://simsontraining.com/wp-content/uploads/2015/03/testimonial_icon-60x60.png" className={styles.Icon} alt="group_icon" />
                </div>
                <div className={styles.Chat}>
                    {/* <div className={styles.Title}>
                        {props.name}
                    </div> */}
                    <div className={styles.Text}>
                        {props.preview}
                    </div>
                </div>
                {/* <div className={styles.Unread}>
                    {props.unread}
                </div> */}
                {/* <div className={styles.Time}>
                    {props.time}
                </div> */}
                {/* <div className={styles.Mark}>
                </div> */}
            </div>
        </NavLink>
    </div>
);

export default ChatsItem;
