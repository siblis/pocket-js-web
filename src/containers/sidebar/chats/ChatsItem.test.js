import React from 'react';
import "../../../../setupTests";
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import ChatsItem from './ChatsItem';

describe('render ChatsItem', () => {
    it('render correctly ChatsItem component', () => {
        const ChatsItemComponent = renderer.create(<Router><ChatsItem/></Router>).toJSON();
        expect(ChatsItemComponent).toMatchSnapshot();
    });
    it('render with props correctrly', () => {  
        const props = {
                id: "5ca0c0eb9839f35abe9d4836",
                preview: "some text"
            },
            ChatsItemComponent = mount(<Router><ChatsItem {...props} /></Router>);
        expect(ChatsItemComponent.find('#ChatsItem').length).toEqual(1);
    });
})