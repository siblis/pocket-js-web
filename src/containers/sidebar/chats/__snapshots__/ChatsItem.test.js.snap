// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ChatsItem render correctly ChatsItem component 1`] = `
<div
  className="Block"
  id="ChatsItem"
  onClick={[Function]}
>
  <a
    aria-current={null}
    href="/chats/undefined"
    onClick={[Function]}
  >
    <div
      className="Item"
    >
      <div
        className="Img"
      >
        <img
          alt="group_icon"
          className="Icon"
          src="http://simsontraining.com/wp-content/uploads/2015/03/testimonial_icon-60x60.png"
        />
      </div>
      <div
        className="Chat"
      >
        <div
          className="Text"
        />
      </div>
    </div>
  </a>
</div>
`;
